src/libraries/Microsoft.Extensions.Logging/src/LoggerMessageAttribute.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Provides information to guide the production of a strongly-typed logging method.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public sealed class LoggerMessageAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LoggerMessageAttribute"/> class
+        /// which is used to guide the production of a strongly-typed logging method.
+        /// </summary>
+        /// <remarks>
```

Comment:
I wonder if this should be on the Attribute class itself, and not the constructor. I would think people would see it more at the class level.